packages/@okta/vuepress-site/docs/guides/configure-native-sso/main/index.md

```diff
@@ -0,0 +1,440 @@
+    "device_secret": "kyLmBvfjMPnMGHtBhW9nKz0tLgJc5xbLay1ujkvOQIk="
+}
+```
+
+## Validate the device secret
+
+Occasionally you may want to verify that the device secret is still valid by using the `/introspect` endpoint.
+
+**Example introspect request**
+
+```bash
+  curl --request POST \
+  --url https://${yourOktaDomain}/oauth2/v1/introspect \
+  --header 'Accept: application/json' \
+  --header 'Content-Type: application/x-www-form-urlencoded' \
+  --data-urlencode 'client_id={client #1 id}' \
+  --data-urlencode 'token={device_secret}' \
+  --data-urlencode 'token_type_hint=device_secret'
+```
+
+**Example introspect response**
+
+```json
+{
+    "active": true,
+    "sid": "102oRgEWx5lRTWHilEoGfed4w",
+    "token_type": "urn:x-oath:params:oauth:token-type:device-secret"
+}
+```
+
+The `/introspect` endpoint returns the `sid` that it's tied to. The same value is present in the ID token. By doing this, you can correlate and identify the ID tokens that are minted with the same device secret.
+
+## Revoke the device secret to end a desktop session
+
+Sometimes you have to end a user's desktop session. When you do that, you are signing the user out of one app as well as all of the other registered apps. To end a desktop session, you must revoke the device secret. The revoke request signs the user out from all of the apps that are a part of the Native SSO flow.
+
+**Example request**
+
+```bash
+curl --location --request POST \
+--url https://${yourOktaDomain}/oauth2/{authorizationServerId}/v1/revoke \
+--header 'Accept: application/json' \
+--header 'Content-Type: application/x-www-form-urlencoded' \
+--data-urlencode 'token={device_secret}' \
+--data-urlencode 'client_id={client #1 id}' \
+```
+
+**Example response**
+
+```
+HTTP/1.1 200 OK
+Content-Type: application/json
+```
+
+After you've revoked the device secret, the corresponding access, refresh, and ID tokens are also revoked for that device. You can verify that the revoke was successful by making a request to the `/introspect` endpoint again for that client. You should receive the following response:
```

Comment:
We don't revoke any tokens when the device secret is revoked, however we invalidate the underlying session that the refresh token is bound to. So when trying to use the refresh token to generate new access tokens, it'll fail because the token depends on the associated session to be valid, which it's not anymore. Additionally, the access token usage will depend on a valid refresh token to work, and since the refresh tokens are not valid anymore, they're also considered invalid. Therefore it makes more sense to phrase it like this;
```suggestion
After you've revoked the device secret, the corresponding access and refresh tokens are invalidated for that device. You can verify that the revoke was successful by introspecting the device secret again for that client. You should receive the following response:
```